model/lifecycle.go

```diff
@@ -737,21 +742,6 @@ func createTasksForBuild(project *Project, buildVariant *BuildVariant, b *build.
 	}
 
 	for _, task := range buildVariant.Tasks {
-		if aliases != nil {
```

Comment:
This is probably a result of me not understanding the problem fully, but don't you still need to have this check in some form for patches? Or are patch tasks already guaranteed to match the provided alias once they reach here?

Reply:
Patch tasks will already matched the provided alias before calling createTasksForBuild, and added dependencies, before constructing the taskNames slice, so using just taskNames is correct as it doesn't filter out dependencies.

For reference, createTasksForBuild is called from [AddTasksToBuild](https://github.com/evergreen-ci/evergreen/blob/master/model/lifecycle.go#L561), where Aliases was already unpopulated, and from [CreateBuildFromVersionNoInsert](https://github.com/evergreen-ci/evergreen/blob/master/model/lifecycle.go#L657), which did populate Aliases, but now we handle it beforehand (in the repotracker, as the other callers of this function don't have aliases).